### DISCUSSION TITLE
Q: Morph Normals

### DISCUSSION THREAD

**Discussion by hypnosnhendricks** [zeux/meshoptimizer#221] [Q&A]
Morph normals are delta's applied to the mesh normal. If the normal flipped ( ex: -1.0 to 1.0 ), the delta would be 2.0. 
How does gltfpack take this into consideration? 

Looking at the code and output, it seems like it is quantizing using the offset/scale calculated from the model bounds, which seems like it would be wrong for models that have bounding boxes smaller than the amount a morph target normal could be.

**Comment by zeux** [comment]
I believe right now morph deltas are assumed to be within -1..1 range for normals & tangents (so flipping isn't handled, the deltas will be clamped).

**Reply by hypnosnhendricks** [reply]
That is what I though. My data definitely has morph normals outside that range...which is why I'm running into weird results.

**Reply by zeux** [reply]
This is probably somewhat similar to #194 in that it's possible to skip quantization on any given delta stream that has out of range values, it just hasn't come up before on the models with morph targets I've looked at.
